lms/djangoapps/grades/management/commands/backfill_grades.py

```diff
@@ -0,0 +1,141 @@
+    validate_dependent_option,
+    parse_course_keys,
+)
+from student.models import CourseEnrollment
+from xmodule.modulestore.django import modulestore
+
+from ... import tasks
+
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Example usage:
+        $ ./manage.py lms backfill_grades --all_courses --settings=devstack
+        $ ./manage.py lms backfill_grades 'edX/DemoX/Demo_Course' --settings=devstack
+    """
+    args = '<course_id course_id ...>'
+    help = 'Computes and persists grade values for all learners in specified courses.'
+
+    def add_arguments(self, parser):
+        """
+        Entry point for subclassed commands to add custom arguments.
+        """
+        parser.add_argument(
+            '--courses',
+            dest='courses',
+            nargs='+',
+            help='List of (space separated) courses that need grades backfilled.',
+        )
+        parser.add_argument(
+            '--all_courses',
+            help='Backfill grades for all courses.',
+            action='store_true',
+            default=False,
+        )
+        parser.add_argument(
+            '--routing_key',
+            dest='routing_key',
+            help='Celery routing key to use.',
+        )
+        parser.add_argument(
+            '--batch_size',
+            help='Maximum number of students to calculate grades for, per celery task.',
+            default=100,
+            type=int,
+        )
+        parser.add_argument(
+            '--start_index',
+            help='Offset from which to start processing enrollments.',
+            default=0,
+            type=int,
+        )
+
+    def handle(self, *args, **options):
+        self._set_log_level(options)
+
+        for course_key in self._get_course_keys(options):
+            self.enqueue_backfill_grades_for_course_tasks(course_key, options)
+
+    def enqueue_backfill_grades_for_course_tasks(self, course_key, options):
+        """
+        Enqueue celery tasks to compute and persist all grades for the
+        specified course, in batches.
+        """
+        enrollment_count = CourseEnrollment.objects.num_enrolled_in(course_key)
+        for offset in six.moves.range(options['start_index'], enrollment_count, options['batch_size']):
+            # If any new enrollments are added after the tasks are fired off, they are already persisting grades.
+            # so there is no need to worry about race conditions.
+            task_options = {'routing_key': options['routing_key']} if options.get('routing_key') else {}
+            result = tasks.backfill_grades_for_course.apply_async(
+                kwargs={
+                    'course_key': six.text_type(course_key),
+                    'offset': offset,
+                    'batch_size': options['batch_size'],
+                },
+                options=task_options,
+            )
+            log.info("Created task {task_id} for {course_key} [{offset}...{end}]".format(
```

Comment:
Please add a prefix here so it's easy to find these log statements in Splunk and in the logs.
I suggest using the same prefix we use elsewhere in the grades layer.